### PR TITLE
Allow UART interrupt priority to be higher

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: lib/uart.toit [hunk 1]
```diff
@@ -74,6 +74,10 @@ class Port implements reader.Reader:
   (Note that pins 16 and 17 are used for PSRAM on some modules, so they cannot
    be used for UART0.)
 
+  Setting a $high_priority increases the interrupt priority to level 3 on the ESP 32.
+    If you do not specify true or false for this argument, the high priority is
+    automatically selected for baud rates of 460800 or above.
```
Comment (mikkeldamsgaard) [inline lib/uart.toit:79]: I dont see the check for baud 460800 in the primitive, is that an error?

File: lib/uart.toit [hunk 2]
```diff
@@ -74,6 +74,10 @@ class Port implements reader.Reader:
   (Note that pins 16 and 17 are used for PSRAM on some modules, so they cannot
    be used for UART0.)
 
+  Setting a $high_priority increases the interrupt priority to level 3 on the ESP 32.
```
Comment (kasperl) [inline lib/uart.toit:77]:
```suggestion
  Setting a $high_priority increases the interrupt priority to level 3 on the ESP32.
```

File: lib/uart.toit [hunk 3]
```diff
@@ -82,12 +86,15 @@ class Port implements reader.Reader:
...
       --parity/int=PARITY_DISABLED
-      --mode/int=MODE_UART:
+      --mode/int=MODE_UART
+      --high_priority/bool?=null:
     if (not tx) and (not rx): throw "INVALID_ARGUMENT"
     if mode == MODE_RS485_HALF_DUPLEX and cts: throw "INVALID_ARGUMENT"
     if not MODE_UART <= mode <= MODE_IRDA: throw "INVALID_ARGUMENT"
 
     tx_flags := (invert_tx ? 1 : 0) + (invert_rx ? 2 : 0)
+    if high_priority or high_priority == null and baud_rate >= 460800:
```
Comment (kasperl) [inline lib/uart.toit:96]:
```suggestion
    if high_priority == null: high_priority = baud_rate >= 460800
    if high_priority:
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Pull request merged**